pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -133,6 +133,11 @@ func (c *controller) ValidateMachineClass(ctx context.Context, classSpec *v1alph
 		return nil, nil, machineutils.ShortRetry, err
 	}
 
+	err = c.validateNodeTemplate(machineClass.NodeTemplate)
+	if err != nil {
+		return nil, nil, machineutils.ShortRetry, err
```

Comment:
before return we can add an error log that node template is not valid.

Reply:
good point.